File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    for (let yellow_height = 1; yellow_height <= Math.sqrt(yellow); yellow_height++) {
        if (yellow % yellow_height !== 0) continue;
        
        const yellow_width = yellow / yellow_height;
        
        if (2 * (yellow_width + yellow_height) + 4 === brown) {
            return [yellow_width + 2, yellow_height + 2];
        }
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    for yellow_height in range(1, int(yellow**0.5) + 1):
        if yellow % yellow_height != 0: continue
        
        yellow_width = yellow / yellow_height
        
        if 2 * (yellow_width + yellow_height) + 4 == brown:
            return [yellow_width + 2, yellow_height + 2]